add premature optimization (#11)
christianscott/christianfscott.com: content/blog/premature-optimization/index.md

Of course, taken to the extreme, this would lead us to some awful tech debt[^2]. As with all one sentence pieces of advice this should be taken with a grain of salt. We should expand this a little bit – of course there are problems that we should solve up front. For example, it's spending at least a little bit of time on the architecture of your app well before your codebase grows to a size where you realize any benefits.

How are these "worth it" problems different from the kinds of problems that aren't worth it? I'm honestly not sure – but problems that arise from *irreversible* decisions (like the architecture of your software) are problems that are worth trying to solve up front.

This still feels a little unsatisfying to me. There are certainly problems that arise from irreversible decisions

With all that in mind, here's the updated but far less snappy saying:

> To decide if a problem is worth solving prematurely (that is, before the problem has arisen), first consider:
> 1) the cost of solving the problem now
> 2) the cost of solving the problem at the time it would arise 

> > 3) the size of the fallout of the problem if it becomes un-solveable

Author: ```suggestion
> 3) the size of the fallout if the problem is not solved
```
